x/escrow/keeper/keeper.go

```diff
@@ -97,11 +129,25 @@ func (k *keeper) AccountDeposit(ctx sdk.Context, id types.AccountID, amount sdk.
 		return err
 	}
 
-	if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(amount)); err != nil {
+	authorizedDepositor, err := sdk.AccAddressFromBech32(obj.Depositor)
+	if err != nil {
 		return err
 	}
 
-	obj.Balance = obj.Balance.Add(amount)
+	switch {
+	case owner.Equals(depositor):
+		if err = k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(amount)); err != nil {
+			return err
+		}
+		obj.Balance = obj.Balance.Add(amount)
+	case authorizedDepositor.Equals(depositor):
+		if err = k.fetchFundsFromDepositor(ctx, owner, depositor, amount); err != nil {
+			return err
+		}
+		obj.Funds = obj.Funds.Add(amount)
+	default:
+		return types.ErrUnauthorizedDepositor
+	}
```

Comment:
Can we use the same code for this and [this](https://github.com/ovrclk/akash/pull/1346/files#diff-7892d8c61ce49375a51d974cb679b570f806d3d9fe774f14dbcc8d0ab76ff05bR82-R92)?

Reply:
Done.

`AccountCreate` can't have the case of `ErrUnauthorizedDepositor`, as whatever is the depositor specified during account creation is the one who is authorized from now on. So, `AccountCreate` has only two cases: depositor being the owner or someone else. While `AccountDeposit` can have three cases: depositor being the owner, being the authorized depositor, or some unauthorized depositor.